tools/src/main/java/io/narayana/db/DBAllocator.java

```diff
@@ -0,0 +1,328 @@
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+/**
+ * @author <a href="mailto:karm@redhat.com">Michal Karm Babacek</a>
+ */
+public class DBAllocator extends Allocator {
+    private static final Logger LOGGER = Logger.getLogger(DBAllocator.class.getName());
+
+    DBAllocator() {
+        // Use getInstance
+    }
+
+    public DB allocateDB(final int expiryMinutes) {
+        // Something between nothing and a business week...
+        if (expiryMinutes < 1 || expiryMinutes > 7200) {
+            throw new IllegalArgumentException("expiryMinutes out of range 1 - 7200");
+        }
+
+        final String projectBuildDirectory = System.getProperty("project.build.directory");
+        if (StringUtils.isEmpty(projectBuildDirectory)) {
+            throw new IllegalArgumentException("project.build.directory must not be empty");
+        }
+        final File driversDir = new File(projectBuildDirectory);
+        if (!driversDir.exists()) {
+            throw new IllegalArgumentException(driversDir.getAbsolutePath() + " must exist");
+        }
+
+        // Timeout for fetching DB driver
+        if (StringUtils.isEmpty(System.getProperty("dballocator.driver.url.timeout.seconds"))) {
+            throw new IllegalArgumentException("dballocator.driver.url.timeout.seconds must not be empty");
+        }
+        final long driverFetchTimeoutSeconds = Long.parseLong(System.getProperty("dballocator.driver.url.timeout.seconds"));
+        if (driverFetchTimeoutSeconds < 0 || driverFetchTimeoutSeconds > 1800) {
+            throw new IllegalArgumentException("driverFetchTimeoutSeconds out of range 0 - 1800");
+        }
+
+        // Fetching driver
+        final String dballocatorDriverUrl = System.getProperty("dballocator.driver.url");
+        if (StringUtils.isEmpty(dballocatorDriverUrl)) {
+            throw new IllegalArgumentException("dballocator.driver.url must not be empty");
+        }
+        final URL driverJarSrc;
+        try {
+            driverJarSrc = new URL(dballocatorDriverUrl);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+            LOGGER.severe(e.getMessage());
+            return null;
+        }
+        final String[] driverJarFilename = dballocatorDriverUrl.split("/");
+        final File driverJarDst = new File(driversDir, driverJarFilename[driverJarFilename.length - 1]);
+        long t = System.currentTimeMillis();
+        // < 1K is a suspicious DB driver jar...
+        while (!fileOK(1024, driverJarDst) && ((System.currentTimeMillis() - t) / 1000 < driverFetchTimeoutSeconds)) {
+            try {
+                LOGGER.finest("Downloading DB driver from " + driverJarSrc.getPath());
+                FileUtils.copyURLToFile(driverJarSrc, driverJarDst);
+            } catch (IOException e) {
+                LOGGER.severe(e.getMessage());
```

Comment:
the exception should be printed as whole in case of error happens, not only error message. The stack normally facilitate finding the cause.

This pattern is used at more places over the commit. I would personally vote for logging the whole stacktrace at most of the places possible/reasonable. If you have some reason to not doing so, I'm happy to understand and discuss it.